jenkins/bootstrap_test.py

```diff
@@ -251,6 +251,12 @@ def testPullRef(self):
         self.assertEqual(bootstrap.pull_ref('123'),
             (['+refs/pull/123/merge'], ['FETCH_HEAD']))
 
+    def testBranchRef(self):
+        self.assertEqual(bootstrap.branch_ref('branch:abcd'),
+            (['branch'], ['abcd']))
+        self.assertEqual(bootstrap.branch_ref('master'),
+            (['master'], ['FETCH_HEAD']))
+
 
 class CheckoutTest(unittest.TestCase):
```

Comment:
Please add a test under IntegrationTest for your expected use case-- I'm not convinced that the current fetching code will do the right thing to make it able to be checked out.